engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/SnapshotObject.java

```diff
@@ -191,7 +191,8 @@ public void processEvent(ObjectInDataStoreStateMachine.Event event) {
             s_logger.debug("Failed to update state:" + e.toString());
             throw new CloudRuntimeException("Failed to update state: " + e.toString());
         } finally {
-            if (event == ObjectInDataStoreStateMachine.Event.OperationFailed) {
+            DataObjectInStore obj = objectInStoreMgr.findObject(this, this.getDataStore());
+            if (event == ObjectInDataStoreStateMachine.Event.OperationFailed && !obj.getState().equals(ObjectInDataStoreStateMachine.State.Destroying)) {
```

Comment:
Is there a possibility for 'obj' to be null? Can you put some comment as to why there is a need to check 'obj' state not in 'Destroying'?